l10n_it_fiscalcode/model/res_partner.py

```diff
@@ -13,7 +13,9 @@ def check_fiscalcode(self):
                 return True
             elif (
                 len(partner.fiscalcode) != 16 and
-                partner.company_type == 'person'
+                partner.company_type == 'person' and
```

Comment:
A little refactoring might help readability:
```
if not partner.fiscalcode:
    # Because it is not mandatory
    return True
elif partner.company_type == 'person':
    # Person case
    if partner.company_name:
        # In E-commerce, if there is company_name, 
        # the user might insert VAT in fiscalcode field.
        # Perform the same check as Company case
        return True
    if len(partner.fiscalcode) != 16:
        # Check fiscalcode of a person
        return False
else:
    # Company case
    return True
```